### ISSUE TITLE
Cannot use full text search

### ISSUE THREAD

**Issue by gilcu3** [kiwix/kiwix-android#1324]
**Describe the bug**

Option for full text search is missing from settings since version 2.4 (version 2.2 works). Zim files are recent (2018). When searching terms, it clearly only searches in titles.

**Expected behavior**

Full text should work as in previous versions.

**Steps to reproduce the behavior:**
1. Search a term

**Environment**
- Version of Kiwix Android : 2.5.1
- Device : Redmi Note 6 Pro
- OS version : Android Pie

**Comment by kelson42** [comment]
I think fulltext should work differently from the suggestion system, like with kiwix-serve. Mixing title search and fulltext search does not seem to be a good approach to me.

**Comment by gilcu3** [comment]
I understand, but the problem is usually when one is making a search, and the term doesn't appear in the title, there is no way to find the corresponding article. In version 2.2 it was possible with Full Text Search enabled.

**Comment by gilcu3** [comment]
Just to wrap up, full text search and title search where mixed in version 2.2, but from version 2.4 and up to version 2.5.1 (current) is it possible to do a full text search?

**Comment by kelson42** [comment]
Like you can see the ticket is still open.

**Comment by gilcu3** [comment]
> Like you can see the ticket is still open.

yeah :(,  but shouldn't old implementation work?

**Comment by AxelBoldt** [comment]
I think this would be an extremely useful feature to have; it's sad that the issue has not been assigned to anyone. Do we know why fulltext search stopped working in version 2.4?

**Comment by mnalis** [comment]
>> Option for full text search is missing from settings since version 2.4 (version 2.2 works). Zim files are recent (2018). When searching terms, it clearly only searches in titles.
>> Do we know why fulltext search stopped working in version 2.4?

> I don't remember the details, but AFAIK it was not working well

Hmmm, it looks more like a bug. [Release notes for 2.4](https://github.com/kiwix/kiwix-android/releases/tag/2.4) indicate that not only should fulltext search work, but that it in fact should work ***better*** with that release (_"NEW: Many improvments in fulltext search"_). 

(And quick search through release notes for newer releases do not mention anything about fulltext search being intentionally removed/disabled either).


> probably it was too slow

Hmmm, was Kiwix doing full-search themselves or did they use (if it exists in a file) fulltext index from `.zim` file? 

I've tested now with 317 MB (2.1GB uncompressed HTML) [openstreetmap-wiki_en_all_nopic_2025-07.zim](https://download.kiwix.org/zim/other/openstreetmap-wiki_en_all_nopic_2025-07.zim), and using `zimsearch openstreetmap-wiki_en_all_nopic_2025-07.zim dizalica` (which peruses full-text index present in that `.zim` file), I get results in 0.013 seconds. Even if my phone was 10 times slower than my laptop (and it is probably better than that), it would still take only a fraction of a second, while **increasing usability a lot**.

---

**TL;DR:** research indicates it is likely a regression bug and not intentional disabling of a feature. It should be looked into.

**Comment by jasontitus** [comment]
It is a bug and causes much confusion - full text search works on Mac OS and iOS but silently doesn't on Android.  I just rebuilt and tested a zim file for an hour or two to try and figure out why it didn't return some results and only then did I go hunting for issues on Git.  If latency is a big issue then we can search titles, display them, and then kickoff the slower full text search w/ a spinner (or even text like 'Searching full text...' if it is expected to be many seconds)

**Comment by yamanq** [comment]
Both iOS and Android depend on the libzim. Based on my understanding of the code, the difference is:
- Android version uses only a [SuggestionSearcher](https://github.com/kiwix/kiwix-android/blob/8bec254959ffd3c35eddb82a23f8fa6daeb1cefd/core/src/main/java/org/kiwix/kiwixmobile/core/reader/ZimFileReader.kt#L187), which searches based on title matches
- iOS version uses both a [SuggestionSearcher](https://github.com/kiwix/kiwix-apple/blob/f068a2ef1b033d4d92810f5fd871eaf78cb1aa63/SwiftUI/Model/SearchOperation/SearchOperation.mm#L153) and a [Searcher](https://github.com/kiwix/kiwix-apple/blob/f068a2ef1b033d4d92810f5fd871eaf78cb1aa63/SwiftUI/Model/SearchOperation/SearchOperation.mm#L116) that does a full-text search.

My recommendation for a solution would be to keep the Title search and add the search results from the text search underneath it with a heading "Full Text Search" with a matching heading above the title search results as well
<img width="539" height="700" alt="Image" src="https://github.com/user-attachments/assets/08d41b0a-7ce2-424c-8127-6142c7c23411" />
